Core/src/TrackFitting/GainMatrixUpdater.cpp

```diff
@@ -58,10 +57,9 @@ std::tuple<double, std::error_code> GainMatrixUpdater::visitMeasurement(
     ACTS_VERBOSE("Gain Matrix K:\n" << K);
 
     if (K.hasNaN()) {
-      error = (direction == Direction::Forward)
-                  ? KalmanFitterError::ForwardUpdateFailed
-                  : KalmanFitterError::BackwardUpdateFailed;  // set to error
-      return false;                                           // abort execution
+      // set to error abort execution
+      error = KalmanFitterError::UpdateFailed;
```

Comment:
I guess this was primarily there for diagnostics?

Reply:
Also (not critical): can we catch this case in the unit test? Sonar reports this as uncovered.

Reply:
Potentially one could count the error cases but that can still be done by the KF itself. If the update is symmetrical I think it is beneficial to reflect that also in the interface.

I added a unit test to check the error case.